Sources/LanguageServerProtocol/WorkspaceSymbols.swift

```diff
@@ -0,0 +1,55 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Request for all symbols that match a certain query string.
+///
+/// This request looks up the canonical occurence of each symbol which has a name that contains the query string.
+/// The list of symbol information is returned
+///
+/// Servers that provide workspace symbol queries should set the `workspaceSymbolProvider` server capability.
+///
+/// - Parameters:
+///   - query: The string that should be looked for in symbols of the workspace.
+///
+/// - Returns: Information about each symbol with a name that contains the query string
+public struct WorkspaceSymbolsRequest: RequestType, Hashable {
+
+  public static let method: String = "workspace/symbol"
+  public typealias Response = [SymbolInformation]?
+
+  /// The document in which to lookup the symbol location.
+  public var query: String
+
+  public init(query: String) {
+    self.query = query
+  }
+}
+
+public struct SymbolInformation: Hashable, ResponseType {
+  public var name: String
+
+  public var kind: SymbolKind
+
+  public var deprecated: Bool
```

Comment:
Should be `Bool?` per the spec.

Reply:
oh yes, and I'll return nil instead of returning false for everything

Reply:
Let me know if you want me to include the deprecated property as part of this change, but it seemed like it could be a lot of work for a small gain.

Reply:
I don't think we even have enough information for that in the index today, so `nil` is fine.